gordo_components/builder/build_model.py

```diff
@@ -32,593 +33,576 @@
-    return that path.
-
-
-    Builds the model if needed, or finds it among already existing models if
-    ``model_register_dir`` is non-None, and we find the model there. If
-    `model_register_dir` is set we will also store the model-location of the generated
-    model there for future use. Think about it as a cache that is never emptied.
-
-    Parameters
-    ----------
-    name: str
-        Name of the model to be built
-    model_config: dict
-        Config for the model. See
-        :func:`gordo_components.builder.build_model.build_model`.
-    data_config: dict
-        Config for the data-configuration. See
-        :func:`gordo_components.builder.build_model.build_model`.
-    metadata: dict
-        Extra metadata to be added to the built models if it is built. See
-        :func:`gordo_components.builder.build_model.build_model`.
-    output_dir: Union[os.PathLike, str]
-        A path to where the model will be deposited if it is built.
-    model_register_dir:
-        A path to a register, see `gordo_components.util.disk_registry`. If this is None
-        then always build the model, otherwise try to resolve the model from the
-        registry.
-    replace_cache: bool
-        Forces a rebuild of the model, and replaces the entry in the cache with the new
-        model.
-    evaluation_config: dict
-        Config for the evaluation. See
-        :func:`gordo_components.builder.build_model.build_model`.
-
-    Returns
-    -------
-    Union[os.PathLike, str]:
-        Path to the model
-    """
-    cache_key = calculate_model_key(
-        name, model_config, data_config, evaluation_config, metadata=metadata
-    )
-    if model_register_dir:
-        logger.info(
-            f"Model caching activated, attempting to read model-location with key "
-            f"{cache_key} from register {model_register_dir}"
-        )
-        if replace_cache:
-            logger.info("replace_cache activated, deleting any existing cache entry")
-            disk_registry.delete_value(model_register_dir, cache_key)
+
+        # Check that the model is actually there
+        if existing_model_location and Path(existing_model_location).exists():
+            logger.debug(
+                f"Found existing model at path {existing_model_location}, returning it"
+            )
+            return existing_model_location
+        elif existing_model_location:
+            logger.warning(
+                f"Found that the model-path {existing_model_location} stored in the "
+                f"registry did not exist."
+            )
+            return None
         else:
-            cached_model_location = check_cache(model_register_dir, cache_key)
-            if cached_model_location:
+            logger.info(
+                f"Did not find the model with key {self.cache_key} in the register at "
+                f"{model_register_dir}."
+            )
+            return None
+
+    def build_with_cache(
```

Comment:
I propose we change this so it returns the same as the `build` method (i.e. it loads the model/metadata if it exists in the cache, and returns that model/metadata instead of the location). Maybe they even can just become one function? Then cli.py can become much prettier!

Why was it like this? I think it was because earlier the caching worked by us writing the path to the special ugly file(!), while now we are anyway copying it and stuff.

Reply:
Cool, we'll leave this for the follow up PR. It'll look dapper as dandy Daniel from DecadentDandyVille